File: rambi-engine/src/test/java/com/rambi/MemcacheTest.java
package com.rambi;

import static org.junit.Assert.assertEquals;

import javax.naming.NamingException;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

public class MemcacheTest {

    private TestServer server = new TestServer();

    @Before
    public void init() throws NamingException {
        server.init();
    }

    @Test
    public void testMemcache() throws InterruptedException {
        // PUT on cache
        HttpUtils.put("http://localhost:8080/services/MemcacheTest.js?value=BLA");

        Response response = HttpUtils.get("http://localhost:8080/memcache?key=value");
        assertEquals("BLA", response.getAsString());

        // GET on cache
        response = HttpUtils.get("http://localhost:8080/services/MemcacheTest.js");
        assertEquals("BLA", response.getAsString());

        // DELETE on cache
        HttpUtils.delete("http://localhost:8080/services/MemcacheTest.js");
        response = HttpUtils.get("http://localhost:8080/memcache?key=value");
        assertEquals("", response.getAsString());

        // PUT on cache with expiration
        HttpUtils.put("http://localhost:8080/services/MemcacheTest.js?value=BLA&expirationInMillis=10");

        response = HttpUtils.get("http://localhost:8080/memcache?key=value");
        assertEquals("", response.getAsString());
    }

    @After
    public void end() throws NamingException {
        server.end();
    }
}

File: rambi-engine/src/test/resources/webapp/services/MemcacheTest.js
var cache = importModule('com/rambi/core/memcache.js', 'cache');

var service = {
    get : function(req, resp) {
        var value = cache().get("value");
        resp.print(value);
    },
    put : function(req, resp) {
        if (req.param("expirationInMillis")) {
            var e = parseInt(req.param("expirationInMillis"));
            cache().put("value", req.param("value"), e);
        } else {
            cache().put("value", req.param("value"));
        }
    },
    delete : function(req, resp) {
        cache().delete("value");
    }
};